<?xml version="1.0"?>
<launch>
    <!-- Arguments -->
    <arg name="use_sim_time"    default="false"/>
    <arg name="waypoints_file"  default="$(find-pkg-share waypoint_navigation)/waypoints_cfg/waypoints.yaml"/>
    <arg name="joy_topic"       default="joy"/>
    <arg name="click_topic"     default="clicked_point"/>
    <arg name="save_joy_button" default="5"/> 
    <arg name="world_frame"     default="map"/>
    <arg name="robot_frame"     default="base_footprint"/>
    <arg name="default_rad"     default="1.0"/>


    <!-- Node for saving waypoints -->
    <node pkg="waypoint_navigation" exec="waypoint_saver" output="screen" launch-prefix="xterm -e">
        <param name="filename"          value="$(var waypoints_file)"/>
        <param name="save_joy_button"   value="$(var save_joy_button)"/>
        <param name="world_frame"       value="$(var world_frame)"/>
        <param name="robot_frame"       value="$(var robot_frame)"/>
        <param name="default_rad"       value="$(var default_rad)"/>
        <param name="use_sim_time"      value="$(var use_sim_time)"/>

        <remap from="waypoints_joy"     to="$(var joy_topic)"/>
        <remap from="waypoints_viz"     to="$(var click_topic)"/>
    </node>
</launch>
